frontend/Rhyolite/Frontend/Modal/Base.hs

```diff
@@ -116,6 +116,10 @@ runModalT backdropCfg f = do
     ((a, open), _) <- withModals backdropCfg (getFirst <$> open) $ runEventWriterT (unModalT f)
   pure a
 
+-- | Change the underlying monad of `ModalT`.
+mapModalT :: (Reflex t, MonadHold t m) => (forall x. m x -> n x) -> ModalT t m m a -> ModalT t n n a
+mapModalT f = ModalT . mapEventWriterT f . withEventWriterT ((fmap . fmap) f) . unModalT
```

Comment:
`ModalT t m m` is kind of restrictive. Would it make more sense provide two functions so you could map the underlying monad and also the modal monad separately?

Reply:
Or even two separate map functions...

Reply:
Well, my first impression when using `ModalT` was, that the two monads will be the same, most of the time, so I decided to make it restrictive, as less restricted variants could be added later on.

But, I can of course add them right away. Out of interest: Do we have use cases already, where we have the two monads differ? I mean `runModalT` assumes them to be the same too, for example.

Reply:
Thanks for the quick review, btw.! :-)

Reply:
`runModalT` is intentionally using the same `m` for both because it's job is to "freeze" the modal monad. From that point on, if you want to really know what monad your modal is in, you need `ModalM m`. If it works for you that they are the same most of the time then I guess the extra complexity may not be worth it.

My concern at that point would be making it clear to other users of this function that it's unnecessarily restrictive and they may not actually be able to use this. If you did one of the following I think it would be clear:

  * Document that this is restrictive and it only works when the two monads are the same (which is clear from the types, but I think the intuition is not as obvious)
  * Have it take two functions or similar.

Reply:
Fair enough. I'll adjust accordingly, together with some more improvements.

Reply:
Ok, the improvements ended up being a little more documentation.